### PR TITLE
remove test/internal/eq and test/internal/throws modules

### PR DESCRIPTION
**Pull request by davidchambers**
This pull request replaces all uses of our internal `eq` and `throws` functions with uses of [`assert.deepStrictEqual`][1] and [`assert.throws`][1] respectively.

[1]: https://nodejs.org/api/assert.html#assertdeepstrictequalactual-expected-message
[2]: https://nodejs.org/api/assert.html#assertthrowsfn-error-message

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -82,49 +80,54 @@ def ('$26')
...
+        $.StrMap ($.Unknown),
+        $.String,
+        $.Symbol,
+        $.Type,
+        $.TypeClass,
+        $.Undefined,
+      ]),
+      true
+    );
   });
```
Comment (davidchambers) [inline test/index.js:116]: I had to use `Z.equals` here.

File: test/index.js [hunk 2]
```diff
@@ -1449,95 +1509,96 @@ Since there is no type of which all the above values are members, the type-varia
...
+    eq (($.Array (a)).supertypes, []);
   });
 
   test ('provides the "Array0" type', () => {
-    eq ($.Array0.name) ('Array0');
-    eq ($.Array0.url) (`https://github.com/sanctuary-js/sanctuary-def/tree/v${version}#Array0`);
-    eq ($.Array0.supertypes) ([$.Array ($.Unknown)]);
+    eq ($.Array0.name, 'Array0');
+    eq ($.Array0.url, `https://github.com/sanctuary-js/sanctuary-def/tree/v${version}#Array0`);
+    eq (Z.equals ($.Array0.supertypes, [$.Array ($.Unknown)]), true);
```
Comment (davidchambers) [inline test/index.js:1556]: I had to use `Z.equals` here.

File: test/index.js [hunk 3]
```diff
@@ -1449,95 +1509,96 @@ Since there is no type of which all the above values are members, the type-varia
...
-    eq (($.Array1 (a)).name) ('Array1');
-    eq (($.Array1 (a)).url) (`https://github.com/sanctuary-js/sanctuary-def/tree/v${version}#Array1`);
-    eq (($.Array1 (a)).supertypes) ([$.Array ($.Unknown)]);
+    eq (typeof $.Array1, 'function');
+    eq ($.Array1.length, 1);
+    eq (show ($.Array1), 'Array1 :: Type -> Type');
+    eq (show ($.Array1 (a)), 'Array1 a');
+    eq (($.Array1 (a)).name, 'Array1');
+    eq (($.Array1 (a)).url, `https://github.com/sanctuary-js/sanctuary-def/tree/v${version}#Array1`);
+    eq (Z.equals (($.Array1 (a)).supertypes, [$.Array ($.Unknown)]), true);
```
Comment (davidchambers) [inline test/index.js:1571]: I had to use `Z.equals` here.

File: test/index.js [hunk 4]
```diff
@@ -1449,95 +1509,96 @@ Since there is no type of which all the above values are members, the type-varia
...
-    eq (($.Array2 (a) (b)).name) ('Array2');
-    eq (($.Array2 (a) (b)).url) (`https://github.com/sanctuary-js/sanctuary-def/tree/v${version}#Array2`);
-    eq (($.Array2 (a) (b)).supertypes) ([$.Array ($.Unknown)]);
+    eq (typeof $.Array2, 'function');
+    eq ($.Array2.length, 1);
+    eq (show ($.Array2), 'Array2 :: Type -> Type -> Type');
+    eq (show ($.Array2 (a) (b)), 'Array2 a b');
+    eq (($.Array2 (a) (b)).name, 'Array2');
+    eq (($.Array2 (a) (b)).url, `https://github.com/sanctuary-js/sanctuary-def/tree/v${version}#Array2`);
+    eq (Z.equals (($.Array2 (a) (b)).supertypes, [$.Array ($.Unknown)]), true);
```
Comment (davidchambers) [inline test/index.js:1588]: I had to use `Z.equals` here.

**Review by Avaq — APPROVED** [review]

**Pull request merged**